src/Microsoft.Azure.WebJobs.Host/Extensions/IJobHostMetadataProvider.cs

```diff
@@ -61,5 +61,12 @@ public interface IJobHostMetadataProvider
         /// </summary>
         /// <param name="output">The <see cref="TextWriter"/> to write the graph to.</param>
         void DebugDumpGraph(TextWriter output);
+
+        /// <summary>
+        /// Gets function metadata.
+        /// </summary>
+        /// <param name="functionName">Name of function to return metadata for.</param>
+        /// <returns></returns>
+        FunctionMetadata GetFunctionMetadata(string functionName);
```

Comment:
This unfortunately would be a breaking change to this public interface. Perhaps we don't need this though. The v1 functions host already has the settings logic for looking for these disable settings. That should be enough, meaning we don't have to flow this info via the metadata provider to the host.